### PR TITLE
ENT-4195: Added API triggered Braze campaign send functionality

### PR DESCRIPTION
**Pull request by irfanuddinahmad**
**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production
    - `test.in` for test requirements
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] [Version](https://github.com/edx/ecommerce-worker/blob/master/setup.py) bumped

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/ecommerce-worker/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [Travis](https://travis-ci.org/github/edx/ecommerce-worker), verify version has been pushed to [PyPI](https://pypi.org/project/edx-ecommerce-worker/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [ecommerce](https://github.com/edx/ecommerce) to upgrade dependencies (including ecommerce-worker)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in ecommerce will look for the latest version in PyPi.
    - Note: the ecommerce-worker constraint in ecommerce **must** also be bumped to the latest version in PyPi.
- [ ] Deploy `ecommerce`
- [ ] Deploy `ecomworker` on GoCD.
    - While some functions in ecommerce-worker are run via ecommerce, others are handled by a standalone AMI and must be
      released via GoCD.

### CODE REVIEW

File: ecommerce_worker/braze/v1/client.py [hunk 1]
```diff
@@ -165,11 +175,11 @@ def _post_request(self, body, endpoint):
             r (requests.Response): The http response object
         """
         self.session.headers.update(
-            {'Authorization': u'Bearer {}'.format(self.rest_api_key), 'Content-Type': 'application/json'}
+            {"Authorization": u"Bearer {}".format(self.rest_api_key), "Content-Type": "application/json"}
```
Comment (moconnell1453) [inline ecommerce_worker/braze/v1/client.py:180]: There are a lot of these changes from single quote to double quote and vice versa. I'm curious why there seems to be a mix. Is this intentional, or should we be using all single quotes or all double quotes?

File: ecommerce_worker/braze/v1/client.py [hunk 2]
```diff
@@ -311,26 +320,25 @@ def send_message(
                 }
         """
         if not email_ids or not subject or not body:
-            raise BrazeClientError("Missing parameters for Braze email")
+            raise BrazeClientError('Missing parameters for Braze email')
```
Comment (moconnell1453) [inline ecommerce_worker/braze/v1/client.py:327]: What is the effect of raising this exception? Does this bubble up to the front-end to display a UI message, or does this end up getting logged as an error?

**Review by moconnell1453 — APPROVED** [review]

File: ecommerce_worker/braze/v1/client.py (hunk 2 above)
Comment (irfanuddinahmad) [reply ecommerce_worker/braze/v1/client.py:327]: Logged as part of the task that uses this client

File: ecommerce_worker/braze/v1/client.py (hunk 1 above)
Comment (irfanuddinahmad) [reply ecommerce_worker/braze/v1/client.py:180]: Intentional. The API messages require double quotes. The alternative would be to use json.dumps

**Review by sameenfatima78 — APPROVED** [review]

**Pull request merged**